sorting/src/sorting.cpp

```diff
@@ -12,6 +13,15 @@ void Sorting::hello_callback_(const std_msgs::msg::String::SharedPtr input_str)
     auto message = std_msgs::msg::String();
     std::sort(input_str->data.begin(), input_str->data.end());
     message.data = input_str->data;
-    RCLCPP_INFO(this->get_logger(), "Publishing: '%s'", message.data.c_str());
-    sorting_pub_->publish(message);
+    if (sorting_active_)
+    {
+        RCLCPP_INFO(this->get_logger(), "Publishing: '%s'", message.data.c_str());
+        sorting_pub_->publish(message);
+    }
+}
+
+void Sorting::toggle_publisher_callback_(const std::shared_ptr<sorting_msgs::srv::ToggleSorting::Request> request, std::shared_ptr<sorting_msgs::srv::ToggleSorting::Response> response)
```

Comment:
Ici par exemple pour enlever le warning "var not used", tu peux ne pas déclarer request dans les paramètres.

i.e.

```cpp
void Sorting::toggle_publisher_callback_(const std::shared_ptr<sorting_msgs::srv::ToggleSorting::Request>, std::shared_ptr<sorting_msgs::srv::ToggleSorting::Response> response)
```

( à changer également dans le header du coup ) 

A terme on acceptera pas les warnings donc bon à savoir